scripts/exit.py

```diff
@@ -4,19 +4,435 @@
 This is a two part script, the first portion requires you to burn an ERC20 asset
 on the Matic Network, and then after the burn is checkpointed on Ethereum mainnet,
 collect the ERC20 asset.
+
+Steps for successful withdrawl:
+
+1. Add Matic ERC20 variable data below
+   - MATIC_ERC20_ASSET_ADDR
+   - BURN_AMOUNT
+2. Call script as follows: `brownie run exit burn_asset_on_matic`
+   - Do this in fork mode first ofcourse
+3. Add the Burn TX ID below
+   - MATIC_BURN_TX_ID
+4. Wait a while up to 30 mins
+5. Call check inclusion function to see if the burn tx has been checkpointed.
+   `brownie run exit is_burn_checkpointed`
+6. Once the burn tx has been checkpointed call the exit function
+   `brownie run exit exit`
+   - Do this in fork mode first ofcourse
+
+To test run: brownie run exit tester --network mainnet
 """
-from brownie import accounts
+import math
+from functools import wraps
+from typing import List, Tuple
+
+import rlp
+from brownie import Contract, accounts, network, web3
 from brownie.project import load as load_project
+from eth_utils import keccak
+from hexbytes import HexBytes
+from trie import HexaryTrie
+from web3.types import BlockData, TxReceipt
+
+from scripts.fetch_deployment_data import main as fetch_deployment_data
+
+PreparedLogs = List[Tuple[bytes, List[bytes], bytes]]
+PreparedReceipt = Tuple[bytes, int, bytes, PreparedLogs]
+
 
 # MUST SET VARIABLES BEFORE BURNING ON MATIC
 MSG_SENDER = accounts.add()
 MATIC_ERC20_ASSET_ADDR = ""
 BURN_AMOUNT = 0
 
+# BURN TX HASH
+MATIC_BURN_TX_ID = "0x4486e398e0f2ca4d00bec85edbb9aff94e7085fa2b5ef18319989d9d8e37152f"
+
+
+def keccak256(value):
+    """Thin wrapper around keccak function."""
+    return HexBytes(keccak(value))
+
 
 def burn_asset_on_matic():
+    """Burn an ERC20 asset on Matic Network"""
     ChildERC20 = load_project("maticnetwork/pos-portal@1.5.2").ChildERC20
     asset = ChildERC20.at(MATIC_ERC20_ASSET_ADDR)
     tx = asset.withdraw(BURN_AMOUNT, {"from": MSG_SENDER})
     print("Burn transaction has been sent.")
     print(f"Visit https://explorer-mainnet.maticvigil.com/tx/{tx.txid} for confirmation")
+
+
+def hot_swap_network(network_name):
```

Comment:
This is very cool :) Could it be tweaked some to allow using the script on the testnet as easy as it is on mainnet?

Brownie should offer first-class support for this sort of behavior :thinking: